### PR TITLE
[FIX] stock: select dest location of a SML

### PR DESCRIPTION
**Pull request by adwid**
On a SML, when setting the done quantity, the onchange may change the
destination location selected by the user

To reproduce the issue:
1. In Settings:
    - Enable "Storage Locations"
    - (Then, ensure "Storage Categories" is disabled)
2. Inventory > Operations Types, edit "Internal Transfers":
    - Enable "Show Detailed Operations"
3. Create an internal transfer IT:
    - From: WH/Stock
    - To: WH/Stock
4. Add a detailed operations to IT:
    - Product: anyone
    - To: WH/Stock/Shelf 1
    - Done: 1

Error: One the field "Done" is modified, the "To" changes and becomes
"WH/Stock". This location should not change (it should be
"WH/Stock/Shelf 1")

When setting the done quantity, an onchange recomputes the destination
location of the SML. To do so, it uses a initial location and searches
among its children. Here is the issue: when "Storage Categories" is
disabled, this initial location should be the one selected by the user.

OPW-2704665

### CODE REVIEW

**Comment by adwid** [comment]
Hi @yhu-odoo 
Here is the fix of the issue we talked about two day ago
Can you review it ?

**Review by yhu-odoo — APPROVED** [review]
lgtm
@amoyaux User complained about putaway rule always give a suggestion when change the qty. this fix restricts this situation only to smart putaway. When users don't use smart putaway, it will behave like the old putaway.

File: addons/stock/models/stock_move_line.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def _onchange_qty_done(self):
     def _onchange_putaway_location(self):
         if not self.id and self.user_has_groups('stock.group_stock_multi_locations') and self.product_id and self.qty_done:
             qty_done = self.product_uom_id._compute_quantity(self.qty_done, self.product_id.uom_id)
-            default_dest_location = self._get_default_dest_location()
+            default_dest_location = self._get_default_dest_location() if self.user_has_groups('stock.group_stock_storage_categories') else self.location_dest_id
```
Comment (amoyaux) [inline addons/stock/models/stock_move_line.py:204]: It should be in _get_default_dest_location itself IMO. So we avoid the same issue with the put in pack

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**